types/pg/index.d.ts

```diff
@@ -79,12 +80,16 @@ export interface QueryResultBase {
     fields: FieldDef[];
 }
 
-export interface QueryResult extends QueryResultBase {
-    rows: any[];
+export interface QueryResultRow {
+	[column: string]: any;
 }
 
-export interface QueryArrayResult extends QueryResultBase {
-    rows: any[][];
+export interface QueryResult<R extends QueryResultRow = any> extends QueryResultBase {
+    rows: R[];
+}
+
+export interface QueryArrayResult<R = any[]> extends QueryResultBase {
```

Comment:
I would type this similar to `QueryArrayConfig`, i.e. `QueryArrayResult<R extends any[] = any[]>`.

Reply:
Good catch, I really should stop doing this stuff when I'm tired